lib/semantics/check-omp-structure.cc

```diff
@@ -246,7 +294,47 @@ void OmpStructureChecker::Enter(const parser::OmpProcBindClause &) {
 void OmpStructureChecker::Enter(const parser::OmpReductionClause &) {
   CheckAllowed(OmpClause::REDUCTION);
 }
-void OmpStructureChecker::Enter(const parser::OmpScheduleClause &) {
+
+bool OmpStructureChecker::ScheduleModifierHasType(
+    const parser::OmpScheduleClause &x,
+    const parser::OmpScheduleModifierType::ModType &type) {
+  const auto &modifier{std::get<0>(x.t)};
+  if (modifier.has_value()) {
+    const auto &modType1{std::get<0>(modifier->t)};
+    const auto &modType2{std::get<1>(modifier->t)};
+    if ((modType1.v).v == type ||
+        (modType2.has_value() && (modType2->v).v == type)) {
+      return true;
+    }
+  }
+  return false;
+}
+void OmpStructureChecker::Enter(const parser::OmpScheduleClause &x) {
   CheckAllowed(OmpClause::SCHEDULE);
+
+  // 2.7 Loop Construct Restriction
+  if (GetContext().directive == OmpDirective::DO) {
+    const auto &kind{std::get<1>(x.t)};
+    const auto &chunk{std::get<2>(x.t)};  // chunk_size
+    if ((kind == parser::OmpScheduleClause::ScheduleType::Runtime ||
+            kind == parser::OmpScheduleClause::ScheduleType::Auto) &&
+        chunk.has_value()) {
+      context_.Say(GetContext().clauseSource,
+          "When SCHEDULE(%s) is specified, "
+          "CHUNK_SIZE must not be specified"_err_en_US,
+          parser::ToUpperCaseLetters(
+              parser::OmpScheduleClause::EnumToString(kind)));
+    }
+
+    if (ScheduleModifierHasType(
```

Comment:
How about calling the function HasScheduleModifierType?  Is that an accurate name?

Reply:
Looks like all the "Has" functions are prefixing "Has", I will change that, thanks.

Reply:
I like having names for predicates that makes them stand out as such, but `HasScheduleModifierType` seems to mean something distinct from `ScheduleModifierHasType`.  The original `ScheduleModifierHasType` reads to me as "Does the schedule modifier have a type?" and the suggested "HasScheduleModifierType" reads as "Does `*this` have a schedule modifier type?"  I don't know which is more accurate.

Reply:
The original `ScheduleModifierHasType` is more accurate in the context. Thanks Peter.